configs/projects/_shared-agent-settings.rb

```diff
@@ -200,7 +200,7 @@
   # break gcc or clang if they try to use the RPATH values we forced.
   proj.setting(:cppflags, "-I#{proj.includedir}")
   proj.setting(:ldflags, "-L#{proj.libdir}")
-  if platform.is_cross_compiled?
+  if platform.architecture == 'arm64'
```

Comment:
I think you will need to be careful her not to break something like [sparc](https://github.com/puppetlabs/puppet-runtime/blob/1a77733cf6bfa804d965e5e93afb662dd48ad982/configs/platforms/solaris-10-sparc.rb#L5), which will rely on this `is_cross_compiled` logic to use the project defined cpp flags.

Reply:
I think this one is ok, because of the `platform.is_macos?` above